From d0aa3d858e02343c803df984504bcb02e3e1aa60 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:48 -0400
Subject: [ethernet] bnx2x: enlarge minimal alignemnt of data offset

Message-id: <1409946615-5862-44-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91584
O-Subject: [RHEL7.1 PATCH 43/70] bnx2x: enlarge minimal alignemnt of data offset
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 9927b51469494d50956978b533d848748b792cef
Author: Dmitry Kravkov <Dmitry.Kravkov@qlogic.com>
Date:   Thu Jun 26 14:31:05 2014 +0300

    bnx2x: enlarge minimal alignemnt of data offset

    This improves the performance of driver on machine with L1_CACHE_SHIFT of at
    most 32 bytes [HW was planned for 64-byte aligned fastpath data].

    Signed-off-by: Dmitry Kravkov <Dmitry.Kravkov@qlogic.com>
    Signed-off-by: Yuval Mintz <Yuval.Mintz@qlogic.com>
    Signed-off-by: Ariel Elior <Ariel.Elior@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x.h b/drivers/net/ethernet/broadcom/bnx2x/bnx2x.h
index 38cd44f..d777fae 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x.h
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x.h
@@ -1509,8 +1509,10 @@ struct bnx2x {
 /* TCP with Timestamp Option (32) + IPv6 (40) */
 #define ETH_MAX_TPA_HEADER_SIZE  72
 
- /* Max supported alignment is 256 (8 shift) */
-#define BNX2X_RX_ALIGN_SHIFT  min(8, L1_CACHE_SHIFT)
+ /* Max supported alignment is 256 (8 shift)
+  * minimal alignment shift 6 is optimal for 57xxx HW performance
+  */
+#define BNX2X_RX_ALIGN_SHIFT  max(6, min(8, L1_CACHE_SHIFT))
 
  /* FW uses 2 Cache lines Alignment for start packet and size
   *
-- 
1.7.1